lib/nimble.phx.gen.template/project.ex

```diff
@@ -29,14 +31,26 @@ defmodule Nimble.Phx.Gen.Template.Project do
 
   def new(opts \\ %{}) do
     %__MODULE__{
-      api_project?: opts[:api] === true,
-      otp_app: Mix.Phoenix.otp_app(),
-      base_module: Mix.Phoenix.base(),
-      base_path: "lib/" <> Atom.to_string(Mix.Phoenix.otp_app()),
-      base_test_path: "test/" <> Atom.to_string(Mix.Phoenix.otp_app()),
-      web_module: Mix.Phoenix.base() <> "Web",
-      web_path: "lib/" <> Atom.to_string(Mix.Phoenix.otp_app()) <> "_web",
-      web_test_path: "test/" <> Atom.to_string(Mix.Phoenix.otp_app()) <> "_web"
+      api_project?: api_project?(opts),
+      web_project?: web_project?(opts),
+      live_project?: live_project?(opts),
+      otp_app: otp_app(),
+      base_module: base_module(),
+      base_path: "lib/" <> Atom.to_string(otp_app()),
+      base_test_path: "test/" <> Atom.to_string(otp_app()),
+      web_module: base_module() <> "Web",
+      web_path: "lib/" <> Atom.to_string(otp_app()) <> "_web",
+      web_test_path: "test/" <> Atom.to_string(otp_app()) <> "_web"
     }
   end
+
+  defp api_project?(opts), do: opts[:api] === true
+
+  defp web_project?(opts), do: opts[:web] === true || opts[:live] === true
+
+  defp live_project?(opts), do: opts[:live] === true
```

Comment:
First adding another param/opts such as `web`, `api` and `live` is rather confusing as it's not an attribute of `Nimble.Phx.Gen.Template.Project`. So there is some indirectionr there.

Second, I think we should try `live` and `api` independently. It's possible for an API to have some live views. It's actually the case in Omise Labs as the integration with third-party payment providers requires to implements some views for redirects. To make things flexible, we should allow for both `web` and `api` projects to have LiveView enabled. So `LiveView` could be treated as an addon instead of a variant.

I understand it would be a major change but I think it's the right approach.

Reply:
An API project having the live view, that is not standard @olivierobert as the API project here means that project is generated by `mix phx.new name --no-html --no-webpack`, with no HTML and Webpack....

In Omise, they created a `Web project` and having only `API` endpoints, that is different with a `no HTML, Webpack` project 😀

We can't generate a `live` project along with `--no-html --no-webpack`

![image](https://user-images.githubusercontent.com/11751745/100975674-fcdec800-3570-11eb-9404-7120ce121f44.png)

Reply:
Okay, I understand and you are right that Omise Labs created a Web project. It's just that I am not fully satisfied with having the current options as it's not that simple to understand all of them. I would prefer to find ways this can be simplified.